pkg/causality/interfaces.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package causality
+
+type (
+	conflictKey = int64
+)
+
+type txnEvent interface {
+	ConflictKeys() []conflictKey
```

Comment:
It should be an ordered slice? Or there could be dead locks I guess.

Reply:
We are not locking individual slots. So this should not be a problem.